src/object/srv_ec_aggregate.c

```diff
@@ -2281,61 +2281,87 @@ agg_iterate_pre_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 	return rc;
 }
 
-struct agg_iv_ult_arg {
+struct ec_agg_ult_arg {
 	struct ec_agg_param	*param;
-	ABT_eventual		eventual;
+	daos_epoch_t		*ec_query_p;
+	uint32_t		tgt_idx;
 };
 
 /* Captures the IV values need for pool and container open. Runs in
  * system xstream.
  */
-static void
-agg_iv_ult(void *arg)
+static	int
+ec_agg_init_ult(void *arg)
 {
-	struct agg_iv_ult_arg	*ult_arg = arg;
+	struct ec_agg_ult_arg	*ult_arg = arg;
 	struct ec_agg_param	*agg_param = ult_arg->param;
+	struct ds_pool		*pool = agg_param->ap_pool_info.api_pool;
 	struct daos_prop_entry	*entry = NULL;
-	daos_prop_t		*prop;
-	int			 rc = 0;
+	daos_prop_t		*prop = NULL;
+	int			 rc;
+
+	rc = ds_cont_ec_eph_insert(agg_param->ap_pool_info.api_pool,
+				   agg_param->ap_pool_info.api_cont_uuid,
+				   ult_arg->tgt_idx, &ult_arg->ec_query_p);
```

Comment:
Assume above 'eph_insert' succeeded but following operation failed, we'll retry agg_param_int later and call 'ec_agg_init_ult' again on the same target? The ce_ref will be leaked at the end.

Reply:
if it is failed, ds_cont_ec_eph_delete will be called, so the ce_ref will be decreased.

Reply:
hmm, ds_cont_ec_eph_delete() will be called no matter if ds_cont_ec_eph_insert() succeeds? so 'eph_delete()' could be called here on error cleanup path to decrease 'ce_ref' grabbed by other target?

Reply:
just checked that for some fail case may drop the ref one more time?
for example, in ec_agg_param_init(), dss_ult_execute(ec_agg_init_ult,...) failed just due to failed create ULT, then ec_agg_param_fini(agg_param); will drop a ref which actually was not taken?